### PR TITLE
feat: meta editor

### LINKED ISSUE #107: Meta Editing
User should be able to edit "basic metadata" via the app.  (title, description, keywords, etc)

Changes should be applied to the working meta on field blur.

### PR DESCRIPTION
**Pull request by chriswhong**
Adds `MetadataEditor` component, with a suite of new form inputs (Closes #107 ):

- `MultiTextInput` - A tag input, which generates an array of strings (for keywords, languages, theme)
- `MultiStructuredInput` - A sortable table input which allows the user to create new rows with a specific schema (for Citations and Contributors)
- `TextAreaInput` - A textarea, for description
- `DropdownInput` - A dropdown that allows the user to choose a license type + url from a predefined list

The existing `Metadata` component is now shown when viewing Meta for a historic commit, `MetadataEditor is shown when looking at fsi-linked meta.

**TODO**

- [x] Wire up an api action - `MetadataEditor` debounces all incoming updates from the form input components, and has a `console.log()` stubbed in where the action call should be. We need to determine what API endpoint to use, then make sure it exists on the backend, then wire it up.  Changes from this component should modify the `meta.json` in a dataset's filesystem integration in near real-time

- [ ] Sortable list `key` bug 🐛 -  (PUNTED) When adding new rows of Contributors or Citations, we have a situation where we are unable to provide a permanent unique key to use for react's `key` prop on each row.  We can't use the index because the rows are sortable.  At the moment we are using the value of the first key in the row's data, which is problematic because it can be edited at any time by the user.

The list will get into an awkward state if two rows exist with the same value as their first property. We may need the backend to append a permanent `_id` property for use as `key`, regardless of the state of other properties.  `_id` would need to persist across multiple api calls to update the working `meta`

- [ ] Add drag and drop  to rearrange tags - (PUNTED) May be out of scope for version 0, but we had to remove the `react-input-tags` because it had an outdated version of react-dnd that was not playing well with the latest version used in `MultiStructuredInput`

### CODE REVIEW

**Comment by chriswhong** [comment]
Ready for review, this PR includes a new `MetadataEditor` Component, `fsiSave()` action, and improvements to the static `Metadata` component.

Changes made on the inputs result in writes to `meta.json` once the input loses focus (or are fired manually for some inputs)

File: app/components/form/MultiStructuredInput.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+      <div className='multi-structured-input-container'>
+        <div className='list'>
+          <div className='multi-structured-input'>
+            <div className='row header-row'>
+              {
+                headerRowValues.map((d: string) => (
+                  <div className='input-column' key={d}>
+                    {/* capitalize it */}
+                    {d.charAt(0).toUpperCase() + d.slice(1)}
```
Comment (ramfox) [inline app/components/form/MultiStructuredInput.tsx]: might make more sense to add some css to 'input-column', so we are sure we aren't mutating anything underneath.

```
   text-transform: capitalize 
```

File: app/components/form/MultiStructuredInputRow.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+interface MultiStructuredInputRowProps {
+  item: User | Citation
+  name: 'contributors' | 'citations'
+  index: number
+  onChange: (index: number, value: User | Citation | null) => void
+  onBlur: () => void
+}
+
+const MultiStructuredInputRow: React.FunctionComponent<MultiStructuredInputRowProps> = (props) => {
```
Comment (ramfox) [inline app/components/form/MultiStructuredInputRow.tsx]: Is this being used anywhere?

File: app/components/FauxLink.tsx [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import * as React from 'react'
+
+const ExternalLink: React.FunctionComponent = ({ children }) =>
```
[the remark this answers is not in the document]
Comment (b5) [reply app/components/FauxLink.tsx]: let's avoid using languages other than English for naming unless we really feel it's justified. `LinkText`?

[the remark this answers is not in the document]
Comment (chriswhong) [reply app/components/FauxLink.tsx]: Changed the component  to `PseudoLink`

File: app/components/Metadata.tsx [hunk 4]
```diff
@@ -36,6 +39,37 @@ const renderURL = (url: string) => (
...
+  )
+}
+
+const renderMultiStructured = (value: User[] | Citation[]) => {
+  if (isUserArray(value)) {
+    return renderArrayItemsTable(value)
+  }
+
+  return renderArrayItemsTable(value)
+}
```
[the remark this answers is not in the document]
Comment (chriswhong) [reply app/components/Metadata.tsx]: This is a typescript guard, both conditions run the same table generator, but TS knows what kind of thing we are passing in and won't yell.

File: app/components/form/DropdownInput.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    onChange,
+    options,
+    placeHolder,
+    tooltipFor
+  } = props
+
+  const labelColor = 'primary'
+
+  const control = (base: React.CSSProperties) => ({
+    ...base,
```
[the remark this answers is not in the document]
Comment (chriswhong) [reply app/components/form/DropdownInput.tsx]: This is `react-select` not giving us classes on its child elements.  It comes with a bright blue border on focus by default, and the only way to disable it is by passing down style in react.  See: https://stackoverflow.com/questions/52614304/react-select-remove-focus-border

File: app/components/form/MultiStructuredInput.tsx (hunk 1 above)
Comment (chriswhong) [reply app/components/form/MultiStructuredInput.tsx]: 👍  Will do

File: app/components/form/MultiStructuredInputRow.tsx (hunk 2 above)
Comment (chriswhong) [reply app/components/form/MultiStructuredInputRow.tsx]: Yes, `MultiStructuredInput` uses it.  Each thing in the array renders a `MultiStructuredInputRow`

**Pull request merged**